realm/src/main/java/io/realm/RealmQuery.java

```diff
@@ -1424,12 +1437,91 @@ public long count() {
      * @throws java.lang.RuntimeException Any other error
      */
     public RealmResults<E> findAll() {
+        checkQueryIsNotReused();
         return new RealmResults<E>(realm, query.findAll(), clazz);
     }
 
     /**
      * Find all objects that fulfill the query conditions and sorted by specific field name.
```

Comment:
Perhaps add something to this async method and all the others that they only work if being called from a looper thread?